utoipa-gen/src/path/property.rs

```diff
@@ -1,59 +1,82 @@
-use std::fmt::Display;
+use std::borrow::Cow;
 
+use proc_macro2::Ident;
 use quote::{quote, ToTokens};
 
-use crate::component_type::{ComponentFormat, ComponentType};
+use crate::{
+    component_type::{ComponentFormat, ComponentType},
+    Type,
+};
 
 /// Tokenizable object property. It is used as a object property for components or as property
 /// of request or response body or response header.
-pub(crate) struct Property<'a, T: Display> {
-    pub(crate) is_array: bool,
-    pub(crate) component_type: ComponentType<'a, T>,
+pub(crate) struct Property<'a> {
+    type_definition: Type<'a>,
```

Comment:
I guess this could also be a reference to the `Type` like `&'a Type<'a>` or `Cow<'a, Type<'a>` so there is no need to clone the Type for the `Property`

Reply:
I think I ran into some weird compile errors when I tried to do that last time, but I'll try again now that this branch is feature complete!

Reply:
I think maybe we can make it `&'a Type<'a>` because we need to produce some owning copies somewhere.

Reply:
Yeah it will do :+1:

Reply:
Okay nice! We can resolve the remaining comments too for now then?

Reply:
Yeah